src/client/renderer/r_stain.c

```diff
@@ -105,28 +105,25 @@ static void R_StainNode(const r_stain_t *stain, const r_bsp_node_t *node) {
 		return;
 	}
 
-	if (node->contents & CONTENTS_MASK_SOLID) {
```

Comment:
This contents check is useless and misleading. Nodes are always `CONTENTS_NODE` (-1), so this always passes. We are not staining leafs; we are staining nodes.